components/RPC-Connection.jsx

```diff
@@ -117,6 +118,32 @@ async function syncData(network, path, setReturnValue) {
 				chainValue = api.toString();
 				break;
 			case "query":
+				switch (pathParameters[2]) {
+					case "validators":
+						const api = await ApiPromise.create({ provider: wsProvider })
+						const [currentValidators, currentEra] = await Promise.all([
+						  api.query.session.validators(),
+						  api.query.staking.currentEra(),
+						]);
+
+
+						const validatorStake = await api.query.staking.erasStakers(currentEra.toString(), currentValidators[0])
+						let validatorMinStake = parseInt(validatorStake['total'].toString())
+						
+
+						for (let i = 1; i < currentValidators.length; i++) {
+							const validatorStake = await api.query.staking.erasStakers(currentEra.toString(), currentValidators[i])
```

Comment:
How many validators does this execute for?  This could be a big bottleneck if there are many validators, as they are synchronous requests.

We don't have an example of this yet but you could fire them all off async and wait for say an array of results to have the expected number of items (or time-out) before proceeding.  Still might not be super fast but it would save considerable time.

Reply:
We have 1K validators in the active set on Kusama and 297 on Polkadot. That is why I intend to refresh the default values every month or so to ensure it reflects the trend with min stake.

Reply:
Could you try that mechanism to speed up things after I create a dedicated component?

Reply:
Yeah we can try running them async.  I'm wondering if the Polkadot JS api has a rate-limiter?  Making 1000+ requests each time the page loads could push that limit if it exists. 

I guess the alternative would be to cache these values in a json file in the repo and have a CI weekly action that updates the json (similar to the status-badge system).  Also not ideal considering we are trying to keep the values more up-to-date.  If we start finding many similar application where large requests are being made for a more complex calculation, a system like that may make more sense.

Reply:
@DrW3RK This also might be useful for retrieving batches of data in a single request: https://polkadot.js.org/docs/api/start/api.query.other#map-keys--entries